src/py21cmmc/likelihood.py

```diff
@@ -1782,3 +1882,136 @@ def computeLikelihood(self, model):
+                PS_limit_ks = PS_limit_ks[~np.isnan(PS_limit_ks)]
+                Nkbins = len(PS_limit_ks)
+                PS_limit_vals = self.data[0]['band' + str(round(band))][field,:Nkbins,1] 
+                PS_limit_vars = self.data[0]['band' + str(round(band))][field,:Nkbins,2]  
+
+                kwf_limit_vals = self.data[0]['kwfband' + str(round(band))]
+                Nkwfbins = len(kwf_limit_vals)
+                PS_limit_wfcs = self.data[0]['wfband' + str(round(band))][field,:Nkbins,:]
+                mask = ~np.isnan(PS_limit_wfcs)
+                PS_limit_wfcs = PS_limit_wfcs[mask].reshape([Nkbins, Nkwfbins])
+                
+                model_ks = model[0]["k"]
+                model_zs = self.redshifts
+                zbin = np.argmin(abs(band - model_zs))
+                ModelPS_val = model[0]["delta"][zbin,:Nkwfbins]
+
+                if np.sum(np.isnan(PS_limit_wfcs)) != 0:
+                    mask1 = ~np.isnan(PS_limit_wfcs)                    
+
+                ModelPS_val_afterWF = np.dot(PS_limit_wfcs, ModelPS_val)
+                # Include emulator error term if present
+                if 'delta_err' in model[0].keys():
+                    ModelPS_val_1sigma_upper_afterWF = np.dot(PS_limit_wfcs, ModelPS_val + model[0]['delta_err'][zbin, :Nkwfbins])
+                    ModelPS_val_1sigma_lower_afterWF = np.dot(PS_limit_wfcs, ModelPS_val - model[0]['delta_err'][zbin, :Nkwfbins])
+                    # The upper and lower errors are very similar usually, so we can just take the mean and use that. 
+                    #Commented out code is if we want to take the asymmetry into account
+                    #error_val_upper = np.sqrt(PS_limit_vars + (0.2*ModelPS_val_afterWF)**2 + (ModelPS_val_1sigma_upper_afterWF-ModelPS_val_afterWF)**2 ) 
+                    #error_val_lower = np.sqrt(PS_limit_vars + (0.2*ModelPS_val_afterWF)**2 + (ModelPS_val_1sigma_lower_afterWF-ModelPS_val_afterWF)**2 )
+                    #mask = PS_limit_vals > ModelPS_val_afterWF
+                    #error_val = error_val_lower
+                    #error_val[mask] = error_val_upper[mask]
+                    mean_err = np.mean([ModelPS_val_1sigma_upper_afterWF-ModelPS_val_afterWF, ModelPS_val_afterWF-ModelPS_val_1sigma_lower_afterWF], axis = 0)
+                    error_val = np.sqrt(PS_limit_vars + (0.2*ModelPS_val_afterWF)**2 + (mean_err)**2 )
+                else:
+                    error_val = np.sqrt(PS_limit_vars + (0.2 * ModelPS_val_afterWF)**2)
+                print('old', np.sqrt(PS_limit_vars + (0.2 * ModelPS_val_afterWF)**2))
+                likelihood = 0.5 + 0.5 * erf( ( PS_limit_vals - ModelPS_val_afterWF ) / (np.sqrt(2) * error_val) ) # another way to write likelihood for 1-side Gaussian
+                likelihood[likelihood <= 0.0] = 1e-50
+                lnl += np.nansum(np.log(likelihood))
+                
+        return lnl
+    
+    @cached_property
```

Comment:
why r u pairing? same as before with luminosity function. I dont think u need to pair under the current structure of how u use the emualtor

Reply:
I think I agree